### PR TITLE
Handle YAML files

### PR DESCRIPTION
**Pull request by antoinm**
This aims to make i18n-check supports YAML i18n files

### CODE REVIEW

File: src/bin/index.ts [hunk 1]
```diff
@@ -100,8 +101,8 @@ const main = async () => {
   let localeFiles: TranslationFile[] = [];
 
   const pattern = isMultiFolders
-    ? `{${localePath.join(",").trim()}}/**/*.json`
-    : `${localePath.join(",").trim()}/**/*.json`;
+    ? `{${localePath.join(",").trim()}}/**/*.{json,yaml,yml}`
+    : `${localePath.join(",").trim()}/**/*.{json,yaml,yml}`;
```
Comment (antoinm) [inline src/bin/index.ts:105]: @busypeoples @steos I went with default support of YAML files, but maybe you would prefer using a command argument to choose the format

**Comment by busypeoples** [comment]
@antoinm Thanks for the PR!
The idea to support YAML files is very useful and makes sense.
From what I understand we can support both `ICU` format and `i18next`.
What we should probably do is also write some CLI tests, that reflect the ones we have for JSON.
One test can validate an ICU YAML file and the other one can validate against the i18next format.

Would you want to write those tests? Otherwise we can add them.

File: src/bin/index.ts (hunk 1 above)
Comment (busypeoples) [reply src/bin/index.ts:105]: @antoinm I think we should go for the default approach without having to specify a command argument.

**Comment by antoinm** [comment]
> @antoinm Thanks for the PR! The idea to support YAML files is very useful and makes sense. From what I understand we can support both `ICU` format and `i18next`. What we should probably do is also write some CLI tests, that reflect the ones we have for JSON. One test can validate an ICU YAML file and the other one can validate against the i18next format.
> 
> Would you want to write those tests? Otherwise we can add them.

I can definitely write the tests.
How would you like me to do it ? 
Should I simply add some yaml files (following the different structures) in the folders under `/translations` ? 
Or should we have a `/translations/json` (with the current content of `/translations`) and a `/translation/yaml` folders ? 

I would go for option 2, so we can keep two separate test suites (1 for JSON and one for YAML).
Since you agreed to the default YAML support approach, I'm afraid mixing both JSON and YAML in the same foldeers might complexify the tests. 

What's you opinion on this ?

**Comment by busypeoples** [comment]
> I can definitely write the tests. How would you like me to do it ? Should I simply add some yaml files (following the different structures) in the folders under `/translations` ? Or should we have a `/translations/json` (with the current content of `/translations`) and a `/translation/yaml` folders ?
> 
> I would go for option 2, so we can keep two separate test suites (1 for JSON and one for YAML). Since you agreed to the default YAML support approach, I'm afraid mixing both JSON and YAML in the same foldeers might complexify the tests.
> 
> What's you opinion on this ?

I would say, let's keep it simple for now and add a `yaml` in the `translations` folder, maybe something like this:

```
- translations
  - yaml
    - basic
      - en.yaml
      - de.yaml 
    - multipleFilesFolderExample (replicates the JSON one)
      - de-DE
        - one.yaml
        - two.yaml
        - three.yaml 
        ...
    - i18NextMessageExamples (replicate the JSON i18NextMessageExamples)
```

We will have to structure the translations folder at some point, but before now and to get some test working - this should be enough. What do you think?

File: src/bin/index.ts [hunk 2]
```diff
@@ -119,21 +120,29 @@ const main = async () => {
...
       name,
+      path,
     });
   });
 
-  fileInfos.forEach(({ file, name, path }) => {
-    const rawContent = JSON.parse(fs.readFileSync(file, "utf-8"));
+  fileInfos.forEach(({ extension, file, name, path }) => {
+    let rawContent;
+    if(extension === "json") {
```
Comment (busypeoples) [inline src/bin/index.ts:140]: We should check against `yaml` here and make the existing `json` the default.

File: src/bin/index.ts [hunk 3]
```diff
@@ -119,21 +120,29 @@ const main = async () => {
...
+      rawContent = JSON.parse(fs.readFileSync(file, "utf-8"));
+    }
+    else {
```
Comment (busypeoples) [inline src/bin/index.ts:143]: This should be formatted, to `} else {`.

File: src/bin/index.ts (hunk 2 above)
Comment (antoinm) [reply src/bin/index.ts]: Done.

File: src/bin/index.ts (hunk 3 above)
Comment (antoinm) [reply src/bin/index.ts]: Done. 
It would be great to include a formatter and linter in the project to avoid issues like this.

File: src/bin/index.test.ts [hunk 4]
```diff
@@ -324,8 +325,228 @@ Found unused keys!
...
-        done();
-      }
-    );
+          done();
+        }
+      );
+    });
+  });
+
+  describe("YAML", () => {
```
Comment (antoinm) [inline src/bin/index.test.ts:334]: @busypeoples  To review this file I suggest you check the "Hide whitespaces" option 
image

I've done a little bit differently from what you suggested [here](https://github.com/lingualdev/i18n-check/pull/17#issuecomment-2564083684)
I didn't create the `i18NextMessageExamples` as it does not seem to be used in the CLI tests. 
Also noticed after pushing that i created the `yaml/basic` folder, but I ended up not using it in the tests, so I guess it should be deleted if you're fine with the current additions. 

Basically, I moved all the CLI test under a sub suite "JSON", and created a "YAML" suite so it looks like this: 
```ts

describe("CLI", () => {
  describe("JSON", () => { ... })
  describe("YAML", () => { ... })
})
```

I duplicated all the JSON tests and adapted them to use the yaml files I created, following the same folder structure, under `translations/yaml`. 

I didn't include the tests about folder exlusion and the one about unused keys.

Comment (busypeoples) [reply src/bin/index.test.ts:334]: Great, thanks @antoinm, I will check the changes.

File: src/bin/index.ts (hunk 3 above)
Comment (busypeoples) [reply src/bin/index.ts]: Yes, that makes a lot of sense. Will create an issue for that one, and add it in the coming days.

File: src/bin/index.test.ts (hunk 4 above)
Comment (busypeoples) [reply src/bin/index.test.ts:334]: > I duplicated all the JSON tests and adapted them to use the yaml files I created, following the same folder structure, under translations/yaml.
> 
> I didn't include the tests about folder exlusion and the one about unused keys.

Makes sense, no need to add the exclusion and the unused keys.

**Comment by busypeoples** [comment]
@antoinm great work, I have no comments left, except if we should update the README in the PR or update in another PR.
The README needs to be updated anyway, as we need to create a contributor list - which is not there yet.

**Review by steos — APPROVED** [review]
lgtm 👍

**Review by busypeoples — APPROVED** [review]

**Comment by busypeoples** [comment]
Thanks again for the great work @antoinm

**Comment by antoinm** [comment]
Thanks for the merge! 
I wasn't quick enough, I should have fixed up the review commits.
